server/controllers/paypal.js

```diff
@@ -16,193 +15,121 @@ const {
-      PaymentMethod
-        .findAndCountAll({
-          where: {
-            service: 'paypal',
-            UserId: req.remoteUser.id
-          }
-        })
-        .then(paymentMethods => cb(null, paymentMethods))
-        .catch(cb);
-    }],
-
-    checkExistingPaymentMethod: ['getExistingPaymentMethod', function(cb, results) {
-      async.each(results.getExistingPaymentMethod.rows, (paymentMethod, cbEach) => {
-        if (!paymentMethod.token) {
-          return paymentMethod.destroy()
-            .then(() => cbEach())
-            .catch(cbEach);
-        }
-
-        paypalAdaptive.preapprovalDetails(paymentMethod.token)
-        .then(response => {
-          if (response.approved === 'false' || new Date(response.endingDate) < new Date()) {
-            paymentMethod.destroy()
-              .then(() => cbEach())
-              .catch(cbEach)
-          } else {
-            cbEach();
-          }
-        })
-        .catch(cbEach)
-      }, cb);
-    }],
-
-    createPaymentMethod: ['checkExistingPaymentMethod', function(cb) {
-      PaymentMethod.create({
-        service: 'paypal',
-        UserId: req.remoteUser.id
-      })
-      .then(paymentMethod => cb(null, paymentMethod))
-      .catch(cb);
-    }],
-
-    createPayload: ['createPaymentMethod', function(cb, results) {
-      const payload = {
-        currencyCode: 'USD',
-        startingDate: new Date().toISOString(),
-        endingDate,
-        returnUrl,
-        cancelUrl,
-        displayMaxTotalAmount: false,
-        feesPayer: 'SENDER',
-        maxTotalAmountOfAllPayments,
-        requestEnvelope: {
-          errorLanguage:  'en_US'
-        },
-        clientDetails: results.createPaymentMethod.id
-      };
-      return cb(null, payload);
-    }],
-
-    callPaypal: ['createPayload', function(cb, results) {
-      paypalAdaptive.preapproval(results.createPayload)
-      .then(response => cb(null, response))
-      .catch(cb)
-    }],
-
-    updatePaymentMethod: ['createPaymentMethod', 'createPayload', 'callPaypal', function(cb, results) {
-      const paymentMethod = results.createPaymentMethod;
-      paymentMethod.token = results.callPaypal.preapprovalKey;
-      paymentMethod.save()
-        .then(paymentMethod => cb(null, paymentMethod))
-        .catch(cb);
-    }]
-
-  }, (err, results) => {
-    if (err) return next(err);
-    res.json(results.callPaypal);
-  });
-
+  const startDate = new Date();
+  const endDate = (req.query.endingDate && (new Date(req.query.endingDate))) || moment().add(1, 'years');
+  
+  const payload = {
+    currencyCode: 'USD', // TODO: figure out if there is a reliable way to specify correct currency for a HOST.
+    startingDate: startDate.toISOString(),
+    endingDate: endDate.toISOString(),
+    returnUrl,
+    cancelUrl,
+    displayMaxTotalAmount: false,
+    feesPayer: 'SENDER',
+    maxAmountPerPayment: 2500.00, // PayPal claims this can go up to $10k without needing additional permissions from them.
```

Comment:
Let's start lower and make sure it works first. We can easily disable old preapprovals and have them go through the flow again on website now.